warp/auth_ldap.py

```diff
@@ -0,0 +1,230 @@
+            connection.start_tls()
+        elif (LDAP_AUTH_USE_LDAPS): 	# LDAPS connection
+            tls_configuration = Tls(validate=LDAP_AUTH_VALIDATE_CERT, version=LDAP_AUTH_TLS_VERSION, ciphers=LDAP_AUTH_CIPHER)
+            server = Server(LDAP_AUTH_SERVER, port=LDAP_AUTH_SERVER_PORT, use_ssl=True, get_info=ALL, tls=tls_configuration)
+            connection = Connection(server=server, authentication=LDAP_AUTH_TYPE, read_only=True, user=bindUser, password=password)
+        else :    			# Plain LDAP connection
+            server = Server(LDAP_AUTH_SERVER, port=LDAP_AUTH_SERVER_PORT, use_ssl=False, get_info=ALL) 
+            print("WARNING: Using LDAP non secure connection: " + server)
+            connection = Connection(server=server, authentication=LDAP_AUTH_TYPE, read_only=True, user=bindUser, password=password)
+        
+        if (connection is None):
+            print("Unnable to connect LDAP server: " + server)
+            return {'bind': False} 
+
+        connection.bind()         
+        # print(f'LDAP bind: {connection.result["description"]}')  # "success" if bind is ok
+        if (connection.result['description'] == "invalidCredentials") :            
+            print("LDAP auth invalidCredentials for ("+username+")")
+            return {'bind': False}
+
+        if (LDAP_MATCHING_RULE_IN_CHAIN):   # Servers supporting LDAP_MATCHING_RULE_IN_CHAIN check is done in groups and nested groups
+            # Check Groups on Active directory
+            for groupMap in LDAP_GROUP_MAP:
+	        # Search on groups and subgroups for 
+                searchString = f'(&(objectclass={LDAP_USER_CLASS})({LDAP_USER_ID_ATTRIBUTE}={escape_filter_chars(username)})({LDAP_USER_GROUPS_ATTRIBUTE}:1.2.840.113556.1.4.1941:={groupMap["ldapGroup"]}))'
+                connection.search(LDAP_SEARCH_BASE, searchString, attributes=[LDAP_USER_NAME_ATTRIBUTE, LDAP_USER_GROUPS_ATTRIBUTE])
+                if (len(connection.entries) == 1) :
+                     userInfo = connection.entries[0]
+                     return {'bind': True, 'name': str(userInfo[LDAP_USER_NAME_ATTRIBUTE]), 'warpGroup': groupMap['warpGroup']}
+            print("User is not in authorithed groups: "+ username)
+            return {'bind': False}
+        else :                              # Servers not supporting LDAP_MATCHING_RULE_IN_CHAIN check is done by users direct groups only
+            searchString = f'(&(objectclass={LDAP_USER_CLASS})({LDAP_USER_ID_ATTRIBUTE}={escape_filter_chars(username)}))'
+            connection.search(LDAP_SEARCH_BASE, searchString, attributes=[LDAP_USER_NAME_ATTRIBUTE, LDAP_USER_GROUPS_ATTRIBUTE])
+            if (len(connection.entries) == 1) :
+                userInfo = connection.entries[0]
+                userGroups = userInfo[LDAP_USER_GROUPS_ATTRIBUTE]
+                groupMapping = next((x for x in LDAP_GROUP_MAP if x['ldapGroup'] in userGroups), None)
+                if (groupMapping != None) :
```

Comment:
```suggestion
                if (groupMapping != None) :
                    print("User " + username + " matched warpGroup " + groupMapping['warpGroup'])
```
Just an informative log line that confirms group matched and user authenticated